astropy/modeling/astro_sherpa.py

```diff
@@ -0,0 +1,470 @@
+
+        Returns
+        -------
+        model_copy : `~astropy.modeling.FittableModel` or a list of models.
+            a copy of the input model with parameters set by the fitter
+        """
+
+        tie_list = []
+        try:
+            self._data = Dataset(models[0].n_inputs, x, y, z, xerr, yerr, zerr)
+        except TypeError:
+            self._data = Dataset(models.n_inputs, x, y, z, xerr, yerr, zerr)
+
+        if self._data.ndata > 1:
+            if len(models) == 1:
+                self._fitmodel = ConvertedModel([models.copy() for _ in xrange(self._data.ndata)], tie_list)
+                # Copy the model so each data set has the same model!
+            elif len(models) == self._data.ndata:
+                self._fitmodel = ConvertedModel(models, tie_list)
+            else:
+                raise Exception("Don't know how to handle multiple models unless there is one foreach dataset")
+        else:
+            if len(models) > 1:
+                self._data.make_simfit(len(models))
+                self._fitmodel = ConvertedModel(models, tie_list)
+            else:
+                self._fitmodel = ConvertedModel(models)
+
+        self._fitter = Fit(self._data.data, self._fitmodel.sherpa_model, self._stat_method, self._opt_method, self._est_method, **kwargs)
+        self.fit_info = self._fitter.fit()
+
+        return self._fitmodel.get_astropy_model()
+
+    def est_errors(self, sigma=None, maxiters=None, numcores=1, methoddict=None, parlist=None):
+        """
+        Use sherpa error estimators based on the last fit.
+
+        Parameters:
+            sigma: float
+                this will be set as the confidance interval for which the errors are found too.
+            maxiters: int
+                the maximum number of iterations the error estimator will run before giving up.
+            methoddict: dict
+                !not quite sure couldn't figure this one out yet!
+            parlist: list
+                a list of parameters to find the confidance interval of if none are provided all free
+                parameters will be estimated.
+        """
+        if self._fitter is None:
+            ValueError("Must complete a valid fit before errors can be calculated")
+        if sigma is not None:
+            self._fitter.estmethod.config['sigma'] = sigma
+        if maxiters is not None:
+            self._fitter.estmethod.config['maxiters'] = maxiters
+        if 'numcores' in self._fitter.estmethod.config:
+            if not numcores == self._fitter.estmethod.config['numcores']:
+                self._fitter.estmethod.config['numcores'] = numcores
+
+        return self._fitter.est_errors(methoddict=methoddict, parlist=parlist)
+
+
+class SherpaWrapper(object):
+    pass
+
+sherpa_stats = {'chi2': Chi2, 'chi2constvar': Chi2ConstVar, 'chi2datavar': Chi2DataVar, 'chi2gehrels': Chi2Gehrels, 'chi2modvar': Chi2ModVar, 'chi2xspecvar': Chi2XspecVar, 'leastsq': LeastSq}
+
+
+class Stat(SherpaWrapper):
```

Comment:
+1 to @taldcroft's suggestion if it really makes sense to have a `SherpaWrapper` base class (it is a no-op in the current code). Otherwise I am not sure I see the advantage of that.